src/components/app.jsx

```diff
@@ -51,7 +53,15 @@ export const ApplicationLoading = () => (
 export const Application = ({ conf, dispatch, isFetching, onCritFail, osRelease, reportLinkURL }) => {
     const [storeInitialized, setStoreInitialized] = useState(false);
     const [showStorage, setShowStorage] = useState(false);
+    const [currentStepId, setCurrentStepId] = useState();
     const address = useAddress();
+    const { path } = usePageLocation();
+
+    useEffect(() => {
+        if (path.length === 0 || path[0] !== currentStepId) {
+            setCurrentStepId(path[0]);
```

Comment:
It is expected that in case of empty path currentStepId is set to undefined I think.

Reply:
It works yes, but I realize,maybe it would be cleaner to just omit it though. In that case path[0] !== currentStepId would also work, if path[0] is undefined.